lib/rails_rate_limiter/strategies/sliding_window_log.rb

```diff
@@ -12,12 +12,12 @@ class SlidingWindowLog
       #   requests per time period.
       # @param per [Fixnum, Lambda, Proc] :per Time period in seconds.
       # @param requester [Lambda, Proc] Identifies request
-      # @param client [Object] Redis client
+      # @param client [Object, Proc] Redis client
       def initialize(limit, per, requester, client = nil)
         @limit = limit.respond_to?(:call) ? limit.call : limit
         @expires_in = calculate_expires_in(per)
-        @requester_pattern = compute_requester_pattern(requester)
-        @client = client
+        @requester_pattern = requester
+        @client = client.is_a?(Proc) ? client.call : client # Redis client already responds to #call so we have to strict
```

Comment:
The use case is that, when specifying a redis client instance that is wrapped in a ConnectionPool and we subsequently resize / restart the pool as part of bootstrapping the Rails application, then it will fail when trying to connect to the Redis server. By wrapping it in a Proc, the result gets evaluated at the request level as opposed to outside of the request context, so it will ensure that we will always have a Redis client that we can successfully establish a connection to.

Additionally, being able to accept a Proc also would theoretically also allow an advanced user to dynamically define the Redis client, should they desire to do so.

Reply:
Nice one, thanks! Merging 👍